include/nighthawk/adaptive_load/adaptive_load_controller.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "envoy/common/time.h"
+
+#include "api/adaptive_load/adaptive_load.pb.h"
+#include "api/client/service.grpc.pb.h"
+
+namespace Nighthawk {
+namespace AdaptiveLoad {
```

Comment:
`Nighthawk` would be sufficient to avoid name clashes. The reason I thought we might want `Nighthawk::AdaptiveLoad` is for consistency with existing code where there's `Nighthawk` for common code but `Nighthawk::Client` and `Nighthawk::Server` for other code.

Reply:
Unless we have a good reason (i.e. consistency here buys us something) I would recommend we follow the rule of minimizing the number of namespaces we introduce. This is a fairly small codebase, so we may never need more than one namespace really. Multiple namespaces will pollute the headers (make expressions more verbose) without necessarily adding any value to readability.

Do we have a good reason to introduce the new namespace?

Reply:
my 2p: a convention where namespaces follow directory layout is convenient (which seems to be the case here as-is, I'm leaning towards the way it's proposed right now). It helps making it explicit when components get intertwined. It's also the Envoy way, I think. 
Having said that, a single namespace would also work for me ([Tangentially related issue](https://github.com/envoyproxy/nighthawk/issues/96)).

Reply:
Well, actually I meant to say that it may *actually* be inconvenient in a good way, by forcing the writing-of-the namespace upon the author and making it very visible to reviewers. But thinking more about that, there's also the dependencies in the BUILD file which ought to highlight what's happening, so .. it doesn't really add value. Which would then argue for the single namespace.

Reply:
Thank you for the clarification Otto. What we learned after maintaining a large c++ codebase is that having multiple sub-namespaces adds to the maintenance burden, pollutes headers (as you describe there is more to type and read) and can create "interesting" scenarios if too generic names for sub-namespaces are used. However the last one is not a concern here.

The advise we arrived at is - ideally there is just one namespace per project, unless absolutely necessary. Of course we can deviate from this, but as with any language feature - we should have a good reason. Looks like this may not be the case. Let's see what @eric846 thinks about this.

Reply:
That makes sense, with context I'm leaning towards a single namespace. 
For consistency, I could do a partial follow up on #96 and drop any other namespaces we have, like `Client`?

Reply:
That is a great idea, such PR would be welcome, thank you @oschaaf !

Before we do that - is that going to affect any publicly visible surfaces?

Reply:
If we consider the [include](https://github.com/envoyproxy/nighthawk/tree/master/include/nighthawk) directory with abstract interfaces as such, yes; But other than that, I don't think so.

Reply:
Now changed to a single Nighthawk namespace for these files.